gui/DepthMapFilterOptions.cxx

```diff
@@ -0,0 +1,202 @@
+  connect(d->UI.pushButtonApply, SIGNAL(pressed()),
+          this, SLOT(updateFilters()));
+
+  connect(d->UI.pushButtonReset, SIGNAL(pressed()),
+          this, SLOT(resetFilters()));
+}
+
+//-----------------------------------------------------------------------------
+DepthMapFilterOptions::~DepthMapFilterOptions()
+{
+  QTE_D();
+  d->uiState.save();
+}
+
+//-----------------------------------------------------------------------------
+double DepthMapFilterOptions::getBestCostValueMin()
+{
+  QTE_D();
+
+ return d->UI.doubleSpinBoxBestCostMin->value();
+}
+
+//-----------------------------------------------------------------------------
+double DepthMapFilterOptions::getBestCostValueMax()
+{
+  QTE_D();
+
+ return d->UI.doubleSpinBoxBestCostMax->value();
+}
+
+//-----------------------------------------------------------------------------
+double DepthMapFilterOptions::getUniquenessRatioMin()
+{
+  QTE_D();
+
+ return d->UI.doubleSpinBoxUniquenessRatioMin->value();
+}
+
+//-----------------------------------------------------------------------------
+double DepthMapFilterOptions::getUniquenessRatioMax()
+{
+  QTE_D();
+
+  return d->UI.doubleSpinBoxUniquenessRatioMax->value();
+}
+
+//-----------------------------------------------------------------------------
+void DepthMapFilterOptions::updateFilters()
+{
+  QTE_D();
+
+  double bcMin = d->UI.doubleSpinBoxBestCostMin->value();
+  double bcMax = d->UI.doubleSpinBoxBestCostMax->value();
+  double urMin = d->UI.doubleSpinBoxUniquenessRatioMin->value();
+  double urMax = d->UI.doubleSpinBoxUniquenessRatioMax->value();
+
+  if(bcMax < bcMin)
+  {
+    d->UI.doubleSpinBoxBestCostMax->setValue(bcMin);
+  }
+
+  if(urMax < urMin)
+  {
+    d->UI.doubleSpinBoxUniquenessRatioMax->setValue(urMin);
+  }
+
+  d->UI.doubleSpinBoxBestCostMax->setMinimum(bcMin);
+  d->UI.doubleSpinBoxUniquenessRatioMax->setMinimum(urMin);
+
+  emit filtersChanged();
+
+}
+
+//-----------------------------------------------------------------------------
+void DepthMapFilterOptions::initializeFilters(double bcMin, double bcMax,
+                                              double urMin, double urMax)
+{
+  QTE_D();
+
+  d->initialBCMin = bcMin;
+  d->initialBCMax = bcMax;
+  d->initialURMin = urMin;
+  d->initialURMax = urMax;
+
+  resetFilters();
+}
+
+//-----------------------------------------------------------------------------
+bool DepthMapFilterOptions::isFilterPersist()
+{
+  QTE_D();
+
+  return d->UI.checkBoxPersist->isChecked();
+}
+
+//-----------------------------------------------------------------------------
+void DepthMapFilterOptions::resetFilters()
+{
+  QTE_D();
+
+  QDoubleSpinBox* bcMinSpinBox = d->UI.doubleSpinBoxBestCostMin;
+  QDoubleSpinBox* bcMaxSpinBox = d->UI.doubleSpinBoxBestCostMax;
+  QDoubleSpinBox* urMinSpinBox = d->UI.doubleSpinBoxUniquenessRatioMin;
+  QDoubleSpinBox* urMaxSpinBox = d->UI.doubleSpinBoxUniquenessRatioMax;
+
+  bcMinSpinBox->setValue(d->initialBCMin);
```

Comment:
This logic is wrong; if the current range of the spin box does not include the desired value, it will be clamped to fit the current range. Set the range first (and use `setRange`, please), *then* set the value.